Src/WindowsServer/WindowsServer.Shared/Implementation/AzureHeartbeatProperties.cs

```diff
@@ -0,0 +1,116 @@
+﻿namespace Microsoft.ApplicationInsights.WindowsServer.Implementation
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing;
+
+    internal class AzureHeartbeatProperties
+    {
+        internal static string HeartbeatPropertyPrefix = "azInst_"; // to ensure no collisions with base heartbeat properties
+
+        /// <summary>
+        /// Expected fields extracted from Azure IMS to add to the heartbeat properties. 
+        /// Set as internal for testing.
+        /// </summary>
+        internal readonly List<string> ExpectedAzureImsFields = new List<string>()
+        {
+            "osType",
+            "location",
+            "name",
+            "offer",
+            "platformFaultDomain",
+            "platformUpdateDomain",
+            "publisher",
+            "sku",
+            "version",
+            "vmId",
+            "vmSize",
+            "subscriptionId",
+            "resourceGroupName"
```

Comment:
there should be some user warning/info about collecting these fields in the comment in this Initializer, if we make this pat of default ai.config.

Reply:
I’ve added a comment to the install.xdt file similar to the comment for perf counters.